### ISSUE TITLE
Some mod_snmp counters are not incrementing

### ISSUE THREAD

**Issue by cu12** [proftpd/proftpd#754]
### What I Did

```
  - name: proftpd_maxInstancesLimitCount_total
    oid: 1.3.6.1.4.1.17852.2.2.1.11
    type: counter
    help: Total number of times the MaxInstances limit has been reached - 1.3.6.1.4.1.17852.2.2.1.11
```

Observed in the logs:
```
2018-10-19 06:35:13,857 proftpd-proftpd-7df7f7dc4b-zsv8w proftpd[9] <redact>: MaxInstances (60) reached, new connection denied
2018-10-19 06:35:13,857 proftpd-proftpd-7df7f7dc4b-zsv8w proftpd[9] <redact>: MaxInstances (60) reached, new connection denied
2018-10-19 06:35:13,857 proftpd-proftpd-7df7f7dc4b-zsv8w proftpd[9] <redact>: MaxInstances (60) reached, new connection denied
```
but

```
snmpwalk -v2c -c <community> localhost 1.3.6.1.4.1.17852.2.2.1.11
iso.3.6.1.4.1.17852.2.2.1.11.0 = Counter32: 0
``` 

Some other counters are not working neither such as `1.3.6.1.4.1.17852.2.2.1.8 - proftpd_connectionRefused_total` or `1.3.6.1.4.1.17852.2.2.1.10 - proftpd_segfaultCount_total`

### What I Expected/Wanted

To be able to monitor ProFTPd in a proper way :)

### ProFTPD Version and Configuration

`ProFTPD Version 1.3.6`

snmp configuration:
```
# http://www.proftpd.org/docs/contrib/mod_snmp.html
# The SNMPEngine directive controls whether the mod_snmp will run as an SNMP
# agent, and handle SNMP messages.
SNMPEngine                On

# The SNMPAgent directive configures the mod_snmp module to act as a "master"
# SNMP agent/entity
SNMPAgent                 master 127.0.0.1:161

# The SNMPCommunity directive configures the community string (effectively a
# passphrase) used for authenticating SNMPv1 and SNMPv2 messages.
SNMPCommunity             {{ .Values.proftpd.snmpCommunity }}

# Configure the directory that mod_snmp will use for its database files
SNMPTables                /var/proftpd/snmp
```

```
# Loading modules
LoadModule                 mod_ban.c
LoadModule                 mod_ctrls_admin.c
LoadModule                 mod_sftp.c
LoadModule                 mod_snmp.c
LoadModule                 mod_sql.c
LoadModule                 mod_sql_mysql.c
LoadModule                 mod_sql_passwd.c
LoadModule                 mod_ifsession.c
LoadModule                 mod_proxy_protocol.c

# Including service configuration
IncludeOptions            AllowSymlinks
Include                   /etc/proftpd/ctrls.conf
Include                   /etc/proftpd/ctrls_admin.conf
Include                   /etc/proftpd/ifsession.conf
Include                   /etc/proftpd/ban.conf
Include                   /etc/proftpd/delay.conf
Include                   /etc/proftpd/log.conf
Include                   /etc/proftpd/proxy.conf
Include                   /etc/proftpd/redis.conf
Include                   /etc/proftpd/sftp.conf
Include                   /etc/proftpd/snmp.conf
Include                   /etc/proftpd/sql.conf
```

@Castaglia do you have any clue why this might happen?

**Comment by Castaglia** [comment]
Does an `SNMPLog` reveal any more data, such as errors setting those counters?  And another way to get more information is enable all of the `mod_snmp` trace channels in a `TraceLog`; see the "Logging" section of the [`mod_snmp` docs](http://www.proftpd.org/docs/contrib/mod_snmp.html).

**Comment by Castaglia** [comment]
I see the issue.  Some of the `mod_snmp` event listeners are registered in a _session_ process (_i.e._ after a client connects), but the events (like "MaxInstances reached") are generated _before_ the session is started.

**Comment by Castaglia** [comment]
This should now be fixed in master, and backported to the 1.3.7 branch.  Thanks!
